### ISSUE TITLE
Error stubbing/doubling a scene.

### ISSUE THREAD

**Issue by bitwes** [bitwes/Gut#503]
I converted a project to 4.0 and discovered that a scene I was doubling was generating "You cannot use stub on...because it is not a double."

I started poking around the code and found this comment
```
# Assumes that the first node we encounter that has a root node path, one
# property, and that property is named 'script' is the GDScript for the
# scene.  This could be flawed.
```
It looks like it is "flawed".

I added some prints to `get_scene_script_object` in `utils.gd` and it appears that the "script" property is not always at index 0, as you can see from the output.  It looks like the rest of this is still valid, but it must now loop through the property indexes to try and find "script" instead of just assuming it is at spot 0.

I'm not sure how to create a scene that does not have the "script" property at index zero, so testing this might be difficult.

```
# utils.gd
func get_scene_script_object(scene):
	var state = scene.get_state()
	var to_return = null
	var root_node_path = NodePath(".")
	var node_idx = 0
	
	print('state = ', state)

	while(node_idx < state.get_node_count() and to_return == null):
		print('get_node_path(', node_idx, ') = ', state.get_node_path(node_idx))
#		print('     ', state.get_node_property_count(node_idx))
		for prop_idx in range(state.get_node_property_count(node_idx)):
			print('    * ', state.get_node_property_name(node_idx, prop_idx))
		# Assumes that the first node we encounter that has a root node path, one
		# property, and that property is named 'script' is the GDScript for the
		# scene.  This could be flawed.
		if(state.get_node_path(node_idx) == root_node_path and state.get_node_property_count(node_idx) == 1):
			print('if 1')
			if(state.get_node_property_name(node_idx, 0) == 'script'):
				print('if 2')
				to_return = state.get_node_property_value(node_idx, 0)

		node_idx += 1

	return to_return
```

```
get_node_path(0) = .
    * layout_mode
    * anchors_preset
    * theme
    * script
````

**Comment by bitwes** [comment]
This appears to fix it:
```gdscript
func get_scene_script_object(scene):
	var state = scene.get_state()
	var to_return = null
	var root_node_path = NodePath(".")
	var node_idx = 0

	while(node_idx < state.get_node_count() and to_return == null):
		print('get_node_path(', node_idx, ') = ', state.get_node_path(node_idx))
		# This tries to find the 'script' property for the root node.  This is 
		# where I found it.  I have nothing else to share.
		if(state.get_node_path(node_idx) == root_node_path):
			for prop_idx in range(state.get_node_property_count(node_idx)):
				if(state.get_node_property_name(node_idx, prop_idx) == 'script'):
					to_return = state.get_node_property_value(node_idx, prop_idx)

		node_idx += 1

	return to_return
```

**Comment by bitwes** [comment]
Fixed in 9.1.0
